README.md

```diff
@@ -13,3 +13,139 @@ This is an NWB extension for storing metadata of devices used in optical experim
 
 ---
 This extension was created using [ndx-template](https://github.com/nwb-extensions/ndx-template).
+
+## Entity relationship diagram
+
+```mermaid
+%%{init: {'theme': 'base', 'themeVariables': {'primaryColor': '#ffffff', "primaryBorderColor': '#144E73', 'lineColor': '#D96F32'}}}%%
+classDiagram
+    direction BT
+    class DeviceModel{
+        <<Device>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        model : text, optional
+    }
+    class Indicator{
+        <<Device>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        label : text
+        injection_location : text, optional
+        injection_coordinates_in_mm : numeric, length 3, optional
+    }
+    class Effector{
+        <<Device>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        label : text
+        injection_location : text, optional
+        injection_coordinates_in_mm : numeric, length 3, optional
+    }
+    class OpticalFiber{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        numerical_aperture : numeric, optional
+        core_diameter_in_um : numeric, optional
+    }
+    class ExcitationSource{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        illumination_type : text, optional
+        excitation_wavelength_in_nm : numeric, optional
+        power_in_W : numeric, optional
+        intensity_in_W_per_m2 : numeric, optional
+        exposure_time_in_s : numeric, optional
+    }
+    class PulsedExcitationSource{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        peak_power_in_W : numeric, optional
+        peak_pulse_energy_in_J : numeric, optional
+        pulse_rate_in_Hz : numeric, optional
+    }
+    class Photodetector{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        detector_type : text, optional
+        detected_wavelength_in_nm : numeric, optional
+        gain : numeric, optional
```

Comment:
Is the `gain` in any units?

Reply:
Good point; actually, it depends on the detector type. Maybe we should add an attribute to express the gain unit as text: `gain_unit`.